### PR TITLE
Adds linting and cleans up

### PR DESCRIPTION
**Pull request by abea**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -408,8 +413,9 @@ function processModule(moduleName) {
             return;
           }
         } else if ((statement.expression.type === 'CallExpression') && (get(statement, 'expression.callee.callee.name') === 'require')) {
-          const arguments = get(statement, 'expression.arguments');
-          if ((arguments.length === 2) && (arguments[0].name === 'self') && (arguments[1].name === 'options')) {
+          const args = get(statement, 'expression.arguments');
```
Comment (abea) [inline app.js:416]: `Parsing error: Binding arguments in strict mode`

File: app.js [hunk 2]
```diff
@@ -556,17 +569,18 @@ function processModule(moduleName) {
       }
       return false;
     } else if ((get(init, 'type') === 'ExpressionStatement') && (get(init, 'expression.type') === 'CallExpression') && (get(init, 'expression.callee.object.name') === 'self') && (get(init, 'expression.callee.property.name') === 'addHelpers')) {
-      const arguments = get(init, 'expression.arguments');
-      if (arguments[0].type === 'ObjectExpression') {
-        helpers = arguments[0];
+      const args = get(init, 'expression.arguments');
```
Comment (abea) [inline app.js:572]: `Parsing error: Binding arguments in strict mode`

File: app.js [hunk 3]
```diff
@@ -575,25 +589,25 @@ function processModule(moduleName) {
 
   function onEvent(init) {
     if ((get(init, 'type') === 'ExpressionStatement') && (get(init, 'expression.type') === 'CallExpression') && (get(init, 'expression.callee.object.name') === 'self') && (get(init, 'expression.callee.property.name') === 'on')) {
-      const arguments = get(init, 'expression.arguments');
-      if ((arguments[0].type !== 'Literal') || (arguments[1].type !== 'Literal')) {
+      const args = get(init, 'expression.arguments');
```
Comment (abea) [inline app.js:592]: `Parsing error: Binding arguments in strict mode`

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+version: 2
+jobs:
+  build:
+    docker:
+      - image: circleci/node:8-browsers
```
Comment (boutell) [inline .circleci/config.yml]: node 12!

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+      - image: mongo:3.6.11
```
Comment (boutell) [inline .circleci/config.yml]: I'd go current mongo too.

File: .circleci/config.yml [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    steps:
+      - checkout
+      - run:
+          name: update-npm
+          command: 'sudo npm install -g npm@5'
```
Comment (boutell) [inline .circleci/config.yml]: that is a very old version of npm I believe.

File: app.js (hunk 1 above)
Comment (boutell) [reply app.js:416]: Oh yeah, that name should be avoided, too magical.

**Review by boutell — CHANGES_REQUESTED** [review]
circle config uses old stuff

**Review by boutell — APPROVED** [review]

**Pull request merged**